packages/ripple-binary-codec/src/serdes/binary-parser.ts

```diff
@@ -13,8 +17,13 @@ class BinaryParser {
    * Initialize bytes to a hex string
    *
    * @param hexBytes a hex string
+   * @param definitions Rippled definitions used to parse the values of transaction types and such.
+   *                          Can be customized for sidechains and amendments.
    */
-  constructor(hexBytes: string) {
+  constructor(
+    hexBytes: string,
+    readonly definitions: XrplDefinitionsBase = DEFAULT_DEFINITIONS,
```

Comment:
Why is this readonly?

Reply:
Huh, I guess I thought it had to be since I saw so many readonly's in the other parts of this change - oops! (I don't think it actually needs to be readonly)